integrations/docker/ci-only-images/chip-cirque-device-base/CHIPCirqueDaemon.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+Copyright (c) 2020 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+import subprocess
+import logging
+from multiprocessing.connection import Listener, Client
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+sh = logging.StreamHandler()
+sh.setFormatter(
+    logging.Formatter(
+        '%(asctime)s [%(name)s] %(levelname)s %(message)s'))
+log.addHandler(sh)
+
+
+class ShellCommand:
+    def __init__(self, args=None):
+        self._args = args
+
+    def __call__(self):
+        if not self._args:
+            return [1, "Cannot spwan background process"]
```

Comment:
could we return a class/datastructure instead of a fixed list of "int, string" ? That would make code somewhat more readable and we may also be able to have methods of "success/failure" based on error code 0/1.

The error code seems redundant: it is always 1 on a "error message" and 0 when the error message is empty.

Reply:
Added `CommandStatus` enum and `CommandResponse` struct for holding CirqueDaemon command responses.